projects/plugins/social/src/class-rest-settings-controller.php

```diff
@@ -25,6 +26,24 @@ class REST_Settings_Controller extends WP_REST_Controller {
 	 * @static
 	 */
 	public function register_rest_routes() {
+		register_rest_route(
+			'jetpack/v4',
+			'/social/settings',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'require_admin_privilege_callback' ),
+					'args'                => $this->get_endpoint_args_for_item_schema(),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'require_admin_privilege_callback' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+			)
+		);
```

Comment:
Do you think we should prevent this from being registered if Jetpack is active? We could maybe hook in late on rest_api_init, and see if Jetpack endpoint is available.